### PR TITLE
feat: serialize book reference

### PR DESCRIPTION
**Pull request by ilya-zhuravel**

### CODE REVIEW

File: test/unit/model/reference/reference.utils.test.ts [hunk 1]
```diff
@@ -242,4 +248,30 @@ describe('deserializeReferenceAnnotatedValue', () => {
...
+
+
+  it('should serialize individual author', () => {
+    const xml = serializeReferenceContributorsList('author', [{firstName: 'John', lastName: 'Doe'}]);
+    expect(xmlSerializer.serializeToString(xml)).toBe('<person-group person-group-type="author"><name><given-names>John</given-names><surname>Doe</surname></name></person-group>');
+  });
+
+  it('should serialize group contributor', () => {
+    const xml = serializeReferenceContributorsList('author', [{groupName: 'Teenage mutant ninja turtles'}]);
+    expect(xmlSerializer.serializeToString(xml)).toBe('<person-group person-group-type="author"><name><collab>Teenage mutant ninja turtles</collab></name></person-group>');
```
Comment (Cooryd) [inline test/unit/model/reference/reference.utils.test.ts:269]: Where is the spec for `<collab>` to sit inside of a `<name>` tag? I couldn't find any example of this in the kitchen sink xml

Comment (Cooryd) [reply test/unit/model/reference/reference.utils.test.ts:269]: [this](https://github.com/elifesciences/XML-mapping/blob/7cfdc3919715cb35940ced2f9a6866182bad2698/elife-00666.xml#L1574) and [this](https://github.com/elifesciences/XML-mapping/blob/7cfdc3919715cb35940ced2f9a6866182bad2698/elife-00666.xml#L1654) were the only examples of this in a reference citation

File: src/model/reference/BookReference.ts [hunk 2]
```diff
@@ -25,12 +32,73 @@ export class BookReference extends BackmatterEntity {
...
+    const xml = xmlDoc.createElement('element-citation');
+    xml.setAttribute('publication-type', 'book');
+
+    const edition = xmlDoc.createElement('edition');
+    edition.appendChild(xmlDoc.createTextNode(this.edition));
+    xml.appendChild(edition);
+
+    xml.appendChild(serializeReferenceContributorsList('editor', this.editors));
+
+    const elocation = xmlDoc.createElement('elocation-id');
```
Comment (Cooryd) [inline src/model/reference/BookReference.ts:47]: @FAtherden-eLife is `elocation-id` required on book references? Can't see an example within the kitchen sink or xml example here: https://github.com/libero/editor/issues/98 however it does seem to live on journal references

File: test/unit/model/reference/reference.utils.test.ts (hunk 1 above)
Comment (ilya-zhuravel) [reply test/unit/model/reference/reference.utils.test.ts:269]: This is what Fred explained a contributor structure could be. Some articles have it.

File: src/model/reference/BookReference.ts (hunk 2 above)
Comment (FAtherden-eLife) [reply src/model/reference/BookReference.ts:47]: Hi @Cooryd, good point, it's not in any of the samples, eLife haven't used it before. If we could keep it in, that would be great though, I think there could be a (rare) use-case for ebooks.

File: test/unit/model/reference/reference.utils.test.ts (hunk 1 above)
Comment (FAtherden-eLife) [reply test/unit/model/reference/reference.utils.test.ts:269]: Thanks both. `collab` as child of `person-group` please. It's either `name` (with child elements), or `collab`. 

Example of a book ref w/ `collab` [here](https://elifesciences.org/articles/65066#bib143). xml can be gotten [here](https://elifesciences.org/articles/65066.xml) (search `<ref id="bib143">` after page has loaded).

**Review by Cooryd — APPROVED** [review]

**Pull request merged**